contracts/lbp/LBPWrapper.sol

```diff
@@ -16,121 +16,182 @@ import "openzeppelin-contracts-sol8/token/ERC20/IERC20.sol";
 import "../utils/interface/ILBPFactory.sol";
 import "../utils/interface/IVault.sol";
 import "../utils/interface/ILBP.sol";
+import "hardhat/console.sol"; // <<<<<<<<<<<<<<< Remove
 
 contract LBPWrapper {
-    address public owner;
-    ILBPFactory public LBPFactory;
-    address public lbp;
-    uint256 public swapFeePercentage;
-    bool isInitialized;
-
-    modifier onlyOwner() {
-        require(msg.sender == owner, "LBPWrapper: only owner function");
+    // constants
+    uint256 private constant HUNDRED_PERCENT = 10e18;
+
+    // locked parameter
+    address public admin; // The address of the admin of this contract.
+    address public beneficiary; // The address that recieves fees
+    uint256 public primeDaoFeePercentage; // fee expressed as a % (e.g. 10**18 = 100% fee, toWei('1') = 100%)
+    ILBP public lbp; // address of LBP that is managed by this contract.
+    IERC20[] public tokens; // tokens that are used in the LBP
+    uint256[] public amounts; // amount of tokens that are going to be added as liquidity in LBP
+
+    bool public poolFunded; // true:- LBP is funded; false:- LBP is yet not funded.
+    bool public initialized; // true:- LBP created; false:- LBP not yet created. Makes sure, only initialized once.
+
+    modifier onlyAdmin() {
+        require(msg.sender == admin, "LBPWrapper: only admin function");
         _;
     }
 
     /**
-     * @dev              transfer ownership to new owner
-     * @param _newOwner  new owner address
+     * @dev                             transfer admin rights to new admin
+     * @param _newAdmin                 new admin address
      */
-    function transferOwnership(address _newOwner) public onlyOwner {
+    function transferAdminRights(address _newAdmin) external onlyAdmin {
         require(
-            _newOwner != address(0),
-            "LBPWrapper: new owner cannot be zero"
+            _newAdmin != address(0),
+            "LBPWrapper: new admin cannot be zero"
```

Comment:
I think, we should have a code and in documentation, we can display full error message. Just how balancer, Gnosis does.

Reply:
I agree that this would be a legit approach. The other one would be to squeeze our error messages below the 32 char limit if possible. Maybe we should discuss w/ the team?